server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -2114,6 +2116,10 @@ private File[] buildClassesDex(List<String> jars, File mainDexList) throws Exten
         context.put("engineJars", empty_list);
         context.put("mainDexList", mainDexList.getAbsolutePath());
 
+        // replace parameter name because '--main-dex-list' is deprecated and reported as error
+        // we can't change command format for older version of engine so replace parameter here.
+        // Remove when old versions won't be supported.
+        platformConfig.dxCmd = platformConfig.dxCmd.replace("--main-dex-list", "--main-dex-rules");
```

Comment:
Hacky solution here. `--main-dex-list` is deprecated (see https://github.com/defold/extender/issues/393) and in the latest version it's reported as error. So need to pass `--main-dex-rules` instead. But older engine can't be changed so here we replace argument before pass to template processor.